### ISSUE TITLE
Inconsistent documentation for JavaScript/TypeScript

### ISSUE THREAD

**Issue by sQVe** [kkoomen/vim-doge#54]
**Describe the bug**

The documentation generated for **JavaScript** and **TypeScript** files seems to be inconsistent and somewhat lacking. The following snippets should all provide the same documentation template:

```javascript
/**
 * [TODO:description]
 * @function foo
 * @param {[TODO:type]} bar - [TODO:description]
 */
const foo = function(bar) {
  return true
}

/**
 * [TODO:description]
 * @param {[TODO:type]} bar - [TODO:description]
 */
function foo(bar) {
  return true
}

/**
 * [TODO:description]
 * @param {[TODO:type]} bar - [TODO:description]
 */
export function foo(bar) {
  return true
}

/**
 * [TODO:description]
 * @type {[TODO:type]}
 */
const foo = bar => bar

/**
 * [TODO:description]
 * @type {[TODO:type]}
 */
const foo = bar => {
  return bar
}

/**
 * [TODO:description]
 * @type {[TODO:type]}
 */
export foo = bar => bar
```

Furthermore I would expect the template to include a `return` and that it's a `function`.

**Out of `vim --version`**
```shell
NVIM v0.4.2
Build type: Release
LuaJIT 2.0.5
Compilation: /usr/bin/cc -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -O2 -DNDEBUG -DMIN_LOG_LEVEL=3 -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wshadow -Wconversion -Wmissing-prototypes -Wimplicit-fallthrough -Wvla -fstack-protector-strong -fdiagnostics-color=always -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOURCE -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -I/build/neovim/src/build/config -I/build/neovim/src/neovim-0.4.2/src -I/usr/include -I/build/neovim/src/build/src/nvim/auto -I/build/neovim/src/build/include
Compiled by builduser

Features: +acl +iconv +tui
See ":help feature-compile"

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/usr/share/nvim"

Run :checkhealth for more info
```

**Comment by kkoomen** [comment]
> The following snippets should all provide the same documentation template:

The code you provided is what is currently being generated. Please provide the input and the expected output you would like to see.

> Furthermore I would expect the template to include a return [...]

I can add a `@return` by default when generating a function, if this is what you want. I'd agree with that. 9/10 times you're returning something in a function.

> [...] and that it's a function.

It uses the [`@function`](https://jsdoc.app/tags-function.html) where needed, although I am confirming that some fat-arrow-functions are not documented with this tag, which they should. The `@function` is not being used for functions that are declared using the explicit `function` keyword, since the reason to use `@function` is to mark a regular variable as a function according to the jsdoc documentation.

**Comment by kkoomen** [comment]
I just noticed that these:

```javascript
/**
 * [TODO:description]
 * @type {[TODO:type]}
 */
const foo = bar => bar

/**
 * [TODO:description]
 * @type {[TODO:type]}
 */
const foo = bar => {
  return bar
}

/**
 * [TODO:description]
 * @type {[TODO:type]}
 */
export foo = bar => bar
```

are being parsed wrong. They should indeed match differently and generate different documentation. I will look into this.

**Comment by sQVe** [comment]
@kkoomen Awesome. Tell me if you need any further information from me.

**Comment by sQVe** [comment]
I just noted that `DoGe` does not handle a `class` that extends something with a `.`. See difference here:

```javascript
/**
 * [TODO:description]
 * @extends React
 */
class Component extends SomeComponent {}

/**
 * [TODO:description]
 * @type {[TODO:type]}
 */
class Component extends React.Component {}
```

**Comment by kkoomen** [comment]
>I just noted that DoGe does not handle a class that extends something with a .. See difference here: [...]

@sQVe Ah, nice catch. I forgot those scenarios. I'll fix those immediately as well.

**Comment by kkoomen** [comment]
This bug has been fixed and released in [v1.14.0](https://github.com/kkoomen/vim-doge/tree/v1.14.0).

Feel free to submit any new issues if you experience any unwanted behavior in the future. Thanks for your contribution.

**Comment by sQVe** [comment]
@kkoomen So far a big improvement. I'll keep my eyes open for more inconsistencies. Cheers for the quick fix! :pray: :beers:

**Comment by sQVe** [comment]
@kkoomen This exported rocket function does no longer work now: `export const foo = bar => true
`

**Comment by kkoomen** [comment]
> @kkoomen This exported rocket function does no longer work: export const foo = bar => true 

Interesting you call it a rocket function. I'm used to what the internet calls it: fat-arrow function.

<hr />

This bug has been fixed and released in [v1.14.1](https://github.com/kkoomen/vim-doge/tree/v1.14.1).

Feel free to submit any new issues if you experience any unwanted behavior in the future. Thanks for your contribution.
